src/events/sdk-key-decoder.ts

```diff
@@ -1,27 +1,70 @@
 import { Base64 } from 'js-base64';
 
-const PATH = 'v0/i';
+const EVENT_INGESTION_HOSTNAME_KEY = 'eh';
+const CONFIGURATION_HOSTNAME_KEY = 'ch';
+
+const EVENT_INGESTION_PATH = 'v0/i';
+const ASSIGNMENT_CONFIG_PATH = 'assignment';
+const EDGE_CONFIG_PATH = 'edge';
 
 export default class SdkKeyDecoder {
   /**
    * Decodes and returns the event ingestion hostname from the provided Eppo SDK key string.
    * If the SDK key doesn't contain the event ingestion hostname, or it's invalid, it returns null.
    */
   decodeEventIngestionUrl(sdkKey: string): string | null {
+    return this.decodeHostnames(sdkKey, EVENT_INGESTION_PATH).eventIngestionHostname;
+  }
+
+  /**
+   * Decodes and returns the configuration hostname from the provided Eppo SDK key string.
+   * If the SDK key doesn't contain the configuration hostname, or it's invalid, it returns null.
+   */
+  decodeAssignmentConfigurationUrl(sdkKey: string): string | null {
+    return this.decodeHostnames(sdkKey, ASSIGNMENT_CONFIG_PATH).configurationHostname;
+  }
+
+  /**
+   * Decodes and returns the edge configuration hostname from the provided Eppo SDK key string.
+   * If the SDK key doesn't contain the edge configuration hostname, or it's invalid, it returns null.
+   */
+  decodeEdgeConfigurationUrl(sdkKey: string): string | null {
+    return this.decodeHostnames(sdkKey, EDGE_CONFIG_PATH).configurationHostname;
+  }
```

Comment:
Looking into the future a bit: the clients allow for a `baseUrl` to be passed in; however we will no longer allow this to be `fscdn.eppo.cloud` if a hostname is decoded here as non-null and it will thrown an exception otherwise people will be able to bypass our DNS blocks. Down the line once all SDK keys have rotated or some future date, we can enforce a subdomain to be present and block bare requests to `fscdn.eppo.cloud` in VCL.